labcontrol/db/study.py

```diff
@@ -228,10 +229,18 @@ def samples(self, term=None, limit=None):
             order_by_clause = "order by sample_values->'%s'" % column
 
         if limit is not None:
-            if type(limit) != int:
-                raise ValueError("limit must be an int")
+            # Attempt to cast the limit to an int, and (if that works) verify
+            # that the integer limit is greater than zero
+            try:
+                limit = int(limit)
+            except (ValueError, OverflowError, TypeError):
```

Comment:
Is there a reason to limit this try/catch to particular errors?  Since the only statement wrapped in the try/catch is the cast, and since we aren't returning different error messages based on different error classes, I think it is safe to catch all errors (with a simple `except:`) and raise your custom ValueError.

Reply:
Update—just pushed a commit that changes this to `except Exception:` (PEP8 apparently recommends this form over `except:`, since the latter catches stuff like KeyboardInterrupts).